lib/write.go

```diff
@@ -90,30 +99,46 @@ func (a *ACBuild) Write(output string, overwrite, sign bool, gpgflags []string)
 	gzwriter := gzip.NewWriter(ofile)
 	defer gzwriter.Close()
 
+	// setup tar writer
+	twriter := tar.NewWriter(gzwriter)
+	defer twriter.Close()
+
 	// create the aci writer
-	aw := aci.NewImageWriter(*man, tar.NewWriter(gzwriter))
-	err = filepath.Walk(a.CurrentACIPath, aci.BuildWalker(a.CurrentACIPath, aw, nil))
-	defer aw.Close()
-	if err != nil {
-		pathErr, ok := err.(*os.PathError)
-		if !ok {
-			fmt.Printf("not a path error!\n")
-			return err
-		}
-		syscallErrno, ok := pathErr.Err.(syscall.Errno)
-		if !ok {
-			fmt.Printf("not a syscall errno!\n")
+	switch a.Mode {
+	case BuildModeAppC:
+		man, err := util.GetManifest(a.CurrentImagePath)
+		if err != nil {
 			return err
 		}
-		if pathErr.Op == "open" && syscallErrno != syscall.EACCES {
-			return err
+		aw := aci.NewImageWriter(*man, twriter)
+		err = filepath.Walk(a.CurrentImagePath, aci.BuildWalker(a.CurrentImagePath, aw, nil))
+		defer aw.Close()
+		if err != nil {
+			pathErr, ok := err.(*os.PathError)
+			if !ok {
+				fmt.Printf("not a path error!\n")
+				return err
+			}
+			syscallErrno, ok := pathErr.Err.(syscall.Errno)
+			if !ok {
+				fmt.Printf("not a syscall errno!\n")
```

Comment:
I made a similar comment in https://github.com/containers/build/pull/285#discussion_r96816258. I *think* error wrapping may be preferred here as well, as we just error-return after this.